src/monaca.js

```diff
@@ -2170,15 +2192,67 @@
 
     if(installDependencies.length > 0) {
       process.stdout.write('\n\nInstalling build dependencies...\n');
-
-      deferred.resolve(this._npmInstall(USER_CORDOVA, installDependencies));
+      deferred.resolve(this._npmInstall(projectDir, installDependencies));
     } else {
-      deferred.resolve();
+      deferred.resolve(projectDir);
     }
 
     return deferred.promise;
   };
 
+  Monaca.prototype.getWebpackConfig = function(environment, projectDir) {
+    try {
+      var projectInfoFile = path.resolve(path.join(projectDir, '.monaca', 'project_info.json'));
+      var config = require(projectInfoFile); 
+    } catch(error) {
+      throw 'Failed to require package info.';
+    }
+
+    var framework = config['template-type'];
+    var file = 'webpack.' + environment + '.' + framework +  '.js';
+    var asset = path.resolve(path.join(__dirname, 'template', file));
+
+    if(!fs.existsSync(asset)) {
+      throw 'Failed to locate Webpack config template for framework ' + framework;
+    }
+
+    return fs.readFileSync(asset, 'utf8')
+      .replace(/{{USER_CORDOVA}}/g, this.jsStringEscape(USER_CORDOVA))
+      .replace(/{{PROJECT_DIR}}/g, this.jsStringEscape(projectDir))
+      ;
+  }
+
+  Monaca.prototype.generateBuildConfigs = function(projectDir) {
+    var projectInfoFile = path.resolve(path.join(projectDir, '.monaca', 'project_info.json'));
+    var config = require(projectInfoFile);
+
+    if(!config.build) {
+      return Q.resolve();
```

Comment:
I think we should explicitly return `false` or `null` if we want to tell no build configs need to be generated.